### PR TITLE
feat: add setup_mvn.sh

### PR DESCRIPTION
**Pull request by noondrej96**
Generates settings.xml file for Maven projects for authentication to JFrog

### CODE REVIEW

File: README.md
Comment (polothy) [inline README.md]: Sorry, I don't know maven well so I'm not sure why different repos are needed for snapshots, etc. Should there actually be different Artifactory repos for these? Otherwise, we should just point everyone to fnds-maven and that would handle both uploads/downloads.

Comment (noondrej96) [reply README.md]: The reason why we need 2 repos is because we have to support deployment of releases and snapshots and it has to be separated in `pom.xml` even if it is in the same repository.

We were not sure if deploying to fnds-maven will use fnds-maven-local but we checked it and it works properly so no need for customization. We removed the customization so now it will always use fnds-maven.

File: setup_mvn.sh
Comment (polothy) [inline setup_mvn.sh]:
```suggestion
    IFS=", " read -r -a setting_values <<< "$array"
```

File: setup_mvn.sh
Comment (polothy) [inline setup_mvn.sh]:
```suggestion
    mkdir "${mvn_path}"
```

File: setup_mvn.sh
Comment (polothy) [inline setup_mvn.sh]:
```suggestion
  cat > "$settings_xml" << EOF
```

**Review by polothy — CHANGES_REQUESTED** [review]
So, the remaining thing I'm trying to figure out here is that normally the URLs to artifactory are secrets that we set in GH and Pipeline Forge.

The problem though is that we default to SETUP_MVN to true, but if we are missing the URL secret, it'll exit with an error, which is correct behavior. But, this will break everyone's workflows without modifying them to have the secret or turning SETUP_MVN to false.

This has been a problem with this repo ever since we started adding new package managers. I'm trying to think of a better way forward by either using regular ENV vars or just hard-coding them into this repo (I've avoided that because this is a public repo, but likely that is just silly). I'll try to have an answer for this soon so we can get this merged.

**Comment by polothy** [comment]
Sorry, still trying to gather some feedback.  Hopefully will have something on Monday.

File: setup_mvn.sh
Comment (noondrej96) [reply setup_mvn.sh]: Changed

File: setup_mvn.sh
Comment (noondrej96) [reply setup_mvn.sh]: Changed

File: setup_mvn.sh
Comment (noondrej96) [reply setup_mvn.sh]: Changed

**Review by polothy — APPROVED** [review]
I'm going to approve this to unblock you.  I got the go-ahead to hardcode Artifactory URLs into this repo.

I think the moving forward pattern will be to have defaults to our artifactory repos, but then use ENV vars to override those.

**Comment by polothy** [comment]
Feel free to merge when you want.

**Review by mzapletal91 — APPROVED** [review]

**Pull request merged**
